File: semana_6_patrones_estructurales/codigo_ejemplos/ejemplo1_adapter.py
"""
EJEMPLO 1: Adapter en Python
(Ejemplo del PDF - Semana 6)
"""


class EnchufeEuropeo:
    def conectar(self):
        return "220V conectado"


class Adaptador:
    def __init__(self, enchufe):
        self.enchufe = enchufe
    
    def conectar(self):
        return f"Adaptado a 110V -> {self.enchufe.conectar()}"


e = EnchufeEuropeo()
a = Adaptador(e)
print(a.conectar())

File: semana_6_patrones_estructurales/codigo_ejemplos/ejemplo1_adapter.cpp
/*
 * EJEMPLO 1: Adapter en C++
 * (Version cruzada del ejemplo Python del PDF - Semana 6)
 */

#include <iostream>
#include <string>

using namespace std;

class EnchufeEuropeo {
public:
    string conectar() {
        return "220V conectado";
    }
};

class Adaptador {
private:
    EnchufeEuropeo* enchufe;
    
public:
    Adaptador(EnchufeEuropeo* e) : enchufe(e) {}
    
    string conectar() {
        return "Adaptado a 110V -> " + enchufe->conectar();
    }
};

int main() {
    EnchufeEuropeo e;
    Adaptador a(&e);
    cout << a.conectar() << endl;
    return 0;
}
